pubnub-core/src/tests/mock/transport.rs

```diff
@@ -0,0 +1,37 @@
+use crate::{Message, Timetoken, Transport};
+use async_trait::async_trait;
+use http::Uri;
+use std::future::Future;
+use std::pin::Pin;
+use thiserror::Error;
+
+use mockall::mock;
+
+#[derive(Debug, Error)]
+#[error("mock tranport error")]
+pub struct MockTransportError;
+
+mock! {
+    pub Transport {
+        #[allow(clippy::type_complexity)]
```

Comment:
nitpick: :sob:

Is this really necessary, or was it leftover from development? Some extra type definitions would really clean up this code.

Reply:
You're right! I'll either take care of that at some point or add a comment explicitly justifying the need for that exception.